src/mesh/curve_interpolant.jl

```diff
@@ -0,0 +1,130 @@
+# """
+#     GammaCurve{RealT<:Real, NNODES}
```

Comment:
GammaCurve is the name from Kopriva's book. I think it comes from physicist type notation where Gamma is used to denote the boundary of a domain (as opposed to the mathematical notation \partial\Omega). CurvedEdge would be fine with me, but then calling it an edge here flies in the face of all other notation where we call things surfaces. I actually think PolyCurve is even less descriptive than GammaCurve

Reply:
Also, should this docstring get uncommented?

Reply:
Do you consider `GammaCurve` to be part of the public API? If yes, it should get a docstring. If not, it should only be a comment. I would prefer the second option for now to allow us to work on this without making breaking changes.

Reply:
I agree with keeping the docstring commented. As for the name: "surface" has - for me - a much richer meaning: It implies an object of which it is the surface, or generally that it separates two different regions. AFAICT, `GammaCurve` really has no notion of what it's being used for, except that it is a polynomial that describes a curve over `s` in `[-1, 1]`. `CurvedEdge` would be somewhere in between, but I agree, it's neither here nor their. Maybe just keep using `GammaCurve`.

Reply:
Hmm, maybe `CurvedSurface` then. This leaves it open for the (hopefully) inevitable 3d version of this implementation. Then we could use coordinates rather than x and y and give the coordinates array an NDIMS type parameter but hard set to 2 for now? That way it is like the mesh

Reply:
Sounds good to me! I like the extensibility to 3D, and the name is in agreement with our convention that a surface can be anything (an interface, a boundary, conforming/non-conforming). And to me at least it would be more self-descriptive than `GammaCurve`.

Reply:
I adjusted the naming to `CurvedSurface` but this necessitated a slight rewrite in the mesh file parsing. Michael helped me get this working but it would be good @ranocha for you to have a look. I did some very basic timings between the old mesh construction and the new one and did not see a huge time difference but I am not certain of the "quality" of the constructor